### PR TITLE
giving flexibility to auto merge

### PR DESCRIPTION
**Pull request by eduardacenteno**
Hey @alejoe91 and @DradeAW, @samuelgarcia and I worked together on this change of the auto_merge module to make it more flexible. 

The idea is that the user can choose which steps they want to perform on the data. The motivation was that the auto merge as it is was too strict for multi-units, and would end up not merging them. Therefore, by jumping some steps, we were able to make it work for single and multi units.

### CODE REVIEW

File: spikeinterface/curation/auto_merge.py [hunk 1]
```diff
@@ -89,6 +90,9 @@ def get_potential_auto_merge(
         by default 1.5
     extra_outputs: bool
         If True, an additional dictionary (`outs`) with processed data is returned, by default False
+    steps: None or list of str
+        which steps to run (gives flexibility to running just some steps)
+        If None all steps are done.
```
Comment (alejoe91) [inline spikeinterface/curation/auto_merge.py:95]: can you list the possible steps?

Comment (eduardacenteno) [reply spikeinterface/curation/auto_merge.py:95]: will do!

File: spikeinterface/curation/auto_merge.py [hunk 2]
```diff
@@ -89,6 +90,11 @@ def get_potential_auto_merge(
...
+        Pontential steps: 'min_spike', 'remove_contaminated', 'unit_positions', 'correlogram', 'template_similarity',
```
Comment (alejoe91) [inline spikeinterface/curation/auto_merge.py:96]:
```suggestion
        Pontential steps: 'min_spikes', 'remove_contaminated', 'unit_positions', 'correlogram', 'template_similarity',
```

File: spikeinterface/curation/auto_merge.py [hunk 3]
```diff
@@ -109,54 +115,69 @@ def get_potential_auto_merge(
     #    * auto correlogram is contaminated
     #    * to far away one from each other
 
-    # STEP 1 : 
+    
+    if steps is None:
+        steps = ['min_spike', 'remove_contaminated', 'unit_positions', 'correlogram', 'template_similarity', 'check_increase_score']
```
Comment (alejoe91) [inline spikeinterface/curation/auto_merge.py:120]:
```suggestion
        steps = ['min_spikes', 'remove_contaminated', 'unit_positions', 'correlogram', 'template_similarity', 'check_increase_score']
```

File: spikeinterface/curation/auto_merge.py [hunk 4]
```diff
@@ -110,54 +116,69 @@ def get_potential_auto_merge(
...
```
Comment (alejoe91) [inline spikeinterface/curation/auto_merge.py:121]:
```suggestion
        steps = ['min_spikes', 'remove_contaminated', 'unit_positions', 'correlogram', 'template_similarity', 'check_increase_score']
```

File: spikeinterface/curation/auto_merge.py [hunk 5]
```diff
@@ -90,6 +91,11 @@ def get_potential_auto_merge(
         by default 1.5
     extra_outputs: bool
         If True, an additional dictionary (`outs`) with processed data is returned, by default False
+    steps: None or list of str
+        which steps to run (gives flexibility to running just some steps)
+        If None all steps are done.
+        Pontential steps: 'min_spike', 'remove_contaminated', 'unit_positions', 'correlogram', 'template_similarity',
```
Comment (alejoe91) [inline spikeinterface/curation/auto_merge.py:97]:
```suggestion
        Pontential steps: 'min_spikes', 'remove_contaminated', 'unit_positions', 'correlogram', 'template_similarity',
```

**Review by alejoe91 — APPROVED** [review]

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**